de.dentrassi.pm.aspect.mvnosgi/src/de/dentrassi/pm/aspect/mvnosgi/internal/VirtualizerImpl.java

```diff
@@ -0,0 +1,67 @@
+package de.dentrassi.pm.aspect.mvnosgi.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.osgi.framework.Version;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.dentrassi.pm.aspect.common.osgi.OsgiAspectFactory;
+import de.dentrassi.pm.aspect.virtual.Virtualizer;
+import de.dentrassi.pm.common.ArtifactInformation;
+import de.dentrassi.pm.osgi.bundle.BundleInformation;
+
+public class VirtualizerImpl implements Virtualizer
+{
+    private final static Logger logger = LoggerFactory.getLogger ( VirtualizerImpl.class );
+
+    private final static String GROUP_ID = "groupid";
```

Comment:
Wow you are busy. Unfortunately I cannot keep up with you today. ;-)
There already is. The InterfaceExtender interface allows to add "actions". And in the get actions method you can which if the subject is a channel and use the has aspect method.
This way you can add a button to your own page/controller and implement whatever you need.
On May 5, 2015 18:49, Markus Rathgeb <notifications@github.com> wrote:In de.dentrassi.pm.aspect.mvnosgi/src/de/dentrassi/pm/aspect/mvnosgi/internal/VirtualizerImpl.java:
> +import java.io.InputStream;
> +
> +import org.osgi.framework.Version;
> +import org.slf4j.Logger;
> +import org.slf4j.LoggerFactory;
> +
> +import de.dentrassi.pm.aspect.common.osgi.OsgiAspectFactory;
> +import de.dentrassi.pm.aspect.virtual.Virtualizer;
> +import de.dentrassi.pm.common.ArtifactInformation;
> +import de.dentrassi.pm.osgi.bundle.BundleInformation;
> +
> +public class VirtualizerImpl implements Virtualizer
> +{
> +    private final static Logger logger = LoggerFactory.getLogger ( VirtualizerImpl.class );
> +
> +    private final static String GROUP_ID = "groupid";

The group ID should be configurable.
But this option should only be available if the aspect get's activated.
Could we add configuration to aspects?

—Reply to this email directly or view it on GitHub.